### ISSUE TITLE
Openvpn3 not connecting on Ubuntu 21.10 after upgrade from 20.04

### ISSUE THREAD

**Issue by jwmurray** [OpenVPN/openvpn3-linux#92]
My Openvpn3 client was working  fine under 20.04.  However, it is no longer workng

Systems:  
Ubuntu 21.10 -- Client runing v17 Beta
PFsense server

Connections succeed from mac and windows machines using same profile.

From the Ubuntu Client:

~> openvpn3-admin version --services
OpenVPN 3 D-Bus services:

  - Client backend starter service
     openvpn3-service-backendstart: v17_beta

  - Configuration Service
     openvpn3-service-configmgr:    v17_beta

  - Log Service
     openvpn3-service-logger:       v17_beta

  - Network Configuration Service
     openvpn3-service-netcfg:       v17_beta

  - Session Manager Service
     openvpn3-service-sessionmgr:   v17_beta

~> journalctl --since -30m SYSLOG_IDENTIFIER=net.openvpn.v3.log + SYSLOG_IDENTIFIER=openvpn3-service-logger + SYSLOG_IDENTIFIER=dbus + _SYSTEMD_UNIT=dbus.service + UNIT=dbus.service

Dec 22 22:34:28 xps dbus-daemon[770]: [system] Activating service name='net.openvpn.v3.backends' requested by ':1.163' (uid=1000 pid=33561 comm="openvpn3-admin version --services " label="unconfined") (using servicehelper)
Dec 22 22:34:28 xps net.openvpn.v3.backends[33566]: OpenVPN 3/Linux v17_beta (openvpn3-service-backendstart)
Dec 22 22:34:28 xps net.openvpn.v3.backends[33566]: OpenVPN core 3.git:HEAD:7765540e linux x86_64 64-bit
Dec 22 22:34:28 xps net.openvpn.v3.backends[33566]: Copyright (C) 2012-2020 OpenVPN Inc. All rights reserved.
Dec 22 22:34:28 xps net.openvpn.v3.log[8170]: Logger VERB2: Attached: {tag:18273594641855206253}  [:1.164/net.openvpn.v3.backends]
Dec 22 22:34:28 xps dbus-daemon[770]: [system] Successfully activated service 'net.openvpn.v3.backends'
Dec 22 22:34:28 xps dbus-daemon[770]: [system] Activating service name='net.openvpn.v3.configuration' requested by ':1.163' (uid=1000 pid=33561 comm="openvpn3-admin version --services " label="unconfined") (using servicehelper)
Dec 22 22:34:28 xps net.openvpn.v3.configuration[33571]: OpenVPN 3/Linux v17_beta (openvpn3-service-configmgr)
Dec 22 22:34:28 xps net.openvpn.v3.configuration[33571]: OpenVPN core 3.git:HEAD:7765540e linux x86_64 64-bit
Dec 22 22:34:28 xps net.openvpn.v3.configuration[33571]: Copyright (C) 2012-2020 OpenVPN Inc. All rights reserved.
Dec 22 22:34:28 xps net.openvpn.v3.log[8170]: Logger VERB2: Attached: {tag:1838155090388398190}  [:1.165/net.openvpn.v3.configuration]
Dec 22 22:34:28 xps dbus-daemon[770]: [system] Successfully activated service 'net.openvpn.v3.configuration'
Dec 22 22:34:29 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration VERB1: Device 'tun0' was removed by openvpn
Dec 22 22:34:29 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Virtual device '5dee4522t2a00t432ft8527t845b9d7f7d81' registered on /net/openvpn/v3/netcfg/21673_5dee4522t2a00t432ft8527t845b9d7f7d8>
Dec 22 22:34:29 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Adding IP Adress 192.168.101.34/24 gw 192.168.101.1 ipv6: no
Dec 22 22:34:29 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Setting remote IP address to '173.8.94.220 ipv6: no
Dec 22 22:34:29 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Adding network '172.16.0.0/16' excl: no ipv6: no
Dec 22 22:34:29 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Adding network '20.0.0.0/24' excl: no ipv6: no
Dec 22 22:34:31 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Socket protect called for socket 8, remote: '173.8.94.220', tun: 'tun1', ipv6: no
Dec 22 22:34:32 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration VERB1: Device 'tun1' was removed by openvpn
Dec 22 22:34:32 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Virtual device '5dee4522t2a00t432ft8527t845b9d7f7d81' registered on /net/openvpn/v3/netcfg/21673_5dee4522t2a00t432ft8527t845b9d7f7d8>
Dec 22 22:34:32 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Adding IP Adress 192.168.101.35/24 gw 192.168.101.1 ipv6: no
Dec 22 22:34:32 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Setting remote IP address to '173.8.94.220 ipv6: no
Dec 22 22:34:32 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Ceeonfiguration INFO: Adding network '172.16.0.0/16' excl: no ipv6: no
Dec 22 22:34:32 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Adding network '20.0.0.0/24' excl: no ipv6: no
Dec 22 22:34:34 xps net.openvpn.v3.log[8170]: Logger VERB2: Detached: {tag:18273594641855206253}  [:1.164/net.openvpn.v3.backends]
Dec 22 22:34:35 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Socket protect called for socket 8, remote: '173.8.94.220', tun: 'tun0', ipv6: no
Dec 22 22:34:36 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration VERB1: Device 'tun0' was removed by openvpn
Dec 22 22:34:36 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Virtual device '5dee4522t2a00t432ft8527t845b9d7f7d81' registered on /net/openvpn/v3/netcfg/21673_5dee4522t2a00t432ft8527t845b9d7f7d8>
Dec 22 22:34:36 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Adding IP Adress 192.168.101.37/24 gw 192.168.101.1 ipv6: no
Dec 22 22:34:36 xps net

.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Setting remote IP address to '173.8.94.220 ipv6: no
Dec 22 22:34:36 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Adding network '172.16.0.0/16' excl: no ipv6: no
Dec 22 22:34:36 xps net.openvpn.v3.log[8170]: {tag:2997087250841283571} Network Configuration INFO: Adding network '20.0.0.0/24' excl: no ipv6: no

The log just keeps looking in this manner.  

[logfile.log](https://github.com/OpenVPN/openvpn3-linux/files/7766957/logfile.log)

When I upgraded, I uninstalled openvpn3 and re-installed it using the -impish version.

Any help is greatly appreciated.  Thank you!

**Comment by dsommers** [comment]
Can you verify if this is related to issue #91 ?

**Comment by jwmurray** [comment]
Yes!  This is [#91](https://github.com/OpenVPN/openvpn3-linux/issues/91).

I fixed this by
1. Edited the acceptable algorithms on pfsense to only include AES-256-GCM and ChaCha20-Poly1305
2. Downloaded the updated .ovpn file from pfsense
3. Used the updated .ovpn file 

So, I did not try downgrading to v16-beta.  Rather, I updated the ovpn file so that it does not request that the client openvpn3 use an unacceptable algorithm.

I then looked at my mac client and saw that the mac client is an older version.  I updated the mac client and incorporated the updated ovpn file and the new mac client (version 3.3.3) now connects.  

Thank you!

john
